Uncaught TypeError: Invalid Point: (NaN, 0)

[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.20.0-beta3 x64
**Electron**: 1.6.9
**OS**: linux 4.12.6-1-ARCH
**Thrown From**: [minimap-plus](https://github.com/mehcode/atom-minimap-plus) package 4.29.2


### Stack Trace

Uncaught TypeError: Invalid Point: (NaN, 0)

```
At /usr/share/atom-beta/resources/app/node_modules/text-buffer/lib/point.js:63

TypeError: Invalid Point: (NaN, 0)
    at Function.module.exports.Point.assertValid (/usr/share/atom-beta/resources/app/node_modules/text-buffer/lib/point.js:63:21)
    at DisplayLayer.translateScreenPosition (/usr/share/atom-beta/resources/app/node_modules/text-buffer/lib/display-layer.js:333:17)
    at DisplayLayer.clipScreenPosition (/usr/share/atom-beta/resources/app/node_modules/text-buffer/lib/display-layer.js:391:18)
    at TextEditor.module.exports.TextEditor.clipScreenPosition (/usr/share/atom-beta/resources/app/src/text-editor.js:1794:38)
    at HTMLElement.pixelPositionForScreenPosition (/usr/share/atom-beta/resources/app/src/text-editor-element.js:242:38)
    at StableAdapter.computeScrollTop (/packages/minimap-plus/lib/adapters/stable-adapter.js:60:42)
    at StableAdapter.getScrollTop (/packages/minimap-plus/lib/adapters/stable-adapter.js:51:17)
    at Minimap.getTextEditorScrollRatio (/packages/minimap-plus/lib/minimap.js:598:25)
    at Minimap.getCapedTextEditorScrollRatio (/packages/minimap-plus/lib/minimap.js:610:29)
    at Minimap.getScrollTopFromEditor (/packages/minimap-plus/lib/minimap.js:933:12)
    at Minimap.getScrollTop (/packages/minimap-plus/lib/minimap.js:887:14)
    at Minimap.getLastVisibleScreenRow (/packages/minimap-plus/lib/minimap.js:865:13)
    at Minimap.emitRangeChanges (/packages/minimap-plus/lib/mixins/decoration-management.js:503:38)
    at Minimap.emitDecorationChanges (/packages/minimap-plus/lib/mixins/decoration-management.js:488:10)
    at Minimap.decorateMarker (/packages/minimap-plus/lib/mixins/decoration-management.js:418:10)
    at MinimapHighlightSelected.markerCreated (/packages/minimap-highlight-selected/lib/minimap-highlight-selected.coffee:53:26)
    at MinimapHighlightSelected.markerCreated (/packages/minimap-highlight-selected/lib/minimap-highlight-selected.coffee:1:1)
    at /packages/minimap-highlight-selected/lib/minimap-highlight-selected.coffee:39:62
    at Function.module.exports.Emitter.simpleDispatch (/usr/share/atom-beta/resources/app/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (/usr/share/atom-beta/resources/app/node_modules/event-kit/lib/emitter.js:141:34)
    at /packages/highlight-selected/lib/highlighted-area-view.coffee:163:20
    at ForwardsSingleLine.iterate (/usr/share/atom-beta/resources/app/node_modules/text-buffer/lib/match-iterator.js:94:17)
    at TextBuffer.module.exports.TextBuffer.scanInRange (/usr/share/atom-beta/resources/app/node_modules/text-buffer/lib/text-buffer.js:1033:29)
    at TextEditor.module.exports.TextEditor.scanInBufferRange (/usr/share/atom-beta/resources/app/src/text-editor.js:2737:32)
    at HighlightedAreaView.module.exports.HighlightedAreaView.highlightSelectionInEditor (/packages/highlight-selected/lib/highlighted-area-view.coffee:151:12)
    at /packages/highlight-selected/lib/highlighted-area-view.coffee:136:10
    at Array.forEach (native)
    at HighlightedAreaView.module.exports.HighlightedAreaView.handleSelection (/packages/highlight-selected/lib/highlighted-area-view.coffee:135:27)
    at HighlightedAreaView.handleSelection (/packages/highlight-selected/lib/highlighted-area-view.coffee:1:1)
    at /packages/highlight-selected/lib/highlighted-area-view.coffee:62:8
```

### Commands

```
     -0:42.1.0 core:confirm (input.hidden-input)
  3x -0:42 blur (input.hidden-input)
     -0:39.8.0 intentions:highlight (input.hidden-input)
     -0:39.4.0 fuzzy-finder:toggle-file-finder (input.hidden-input)
  3x -0:39.4.0 blur (input.hidden-input)
     -0:37.9.0 core:confirm (input.hidden-input)
 16x -0:37.9.0 blur (input.hidden-input)
     -0:17.9.0 vim-mode-plus:move-down (input.hidden-input)
     -0:17.7.0 intentions:highlight (input.hidden-input)
  9x -0:17.6.0 editor:move-to-beginning-of-word (input.hidden-input)
     -0:15.6.0 vim-mode-plus:activate-insert-mode (input.hidden-input)
     -0:15.4.0 intentions:highlight (input.hidden-input)
     -0:15.2.0 editor:select-to-end-of-word (input.hidden-input)
     -0:14.5.0 core:backspace (input.hidden-input)
  2x -0:14 core:delete (input.hidden-input)
  4x -0:11.4.0 blur (input.hidden-input)
```

### Non-Core Packages

```
advanced-open-file 0.16.6 
atom-clock 0.1.13 
atom-material-syntax 1.0.6 
atom-material-syntax-dark 0.2.7 
atom-material-ui 2.0.4 
busy-signal 1.4.3 
docblockr 0.11.0 
editorconfig 2.2.2 
ex-mode-hb 0.16.0 
file-icons 2.1.10 
file-watcher 1.1.0 
fuzzy-finder-hb 1.5.8 
highlight-selected 0.13.1 
hyperclick 0.1.3 
intentions 1.1.5 
language-blade 0.34.0 
language-diff 0.7.0 
language-docker 1.1.8 
language-ini 1.19.0 
language-nginx 0.8.0 
language-vue 0.23.1 
less-than-slash 0.17.0 
linter 2.2.0 
linter-eslint 8.2.1 
linter-ui-default 1.6.4 
minimap-find-and-replace 4.5.2 
minimap-hide 0.3.0 
minimap-highlight-selected 4.6.1 
minimap-plus 4.29.2 
minimap-selection 4.5.0 
nord-atom-ui 0.11.0 
northem-dark-atom-ui 2.1.0 
nucleus-dark-ui 0.12.3 
package-sync 1.1.0 
php-cs-fixer 4.1.0 
php-integrator-annotations 1.2.0 
php-integrator-autocomplete-plus 1.6.0 
php-integrator-base 3.0.1 
php-integrator-navigation 1.2.1 
php-integrator-refactoring 1.4.1 
php-twig 4.0.0 
project-manager 3.3.5 
vim-mode-plus 0.96.2 
vim-mode-plus-keymaps-for-surround 0.2.1 
wakatime 7.0.3 
```

I tried to install `minimap` as detailed above but I get the following error.

```
npm ERR! Darwin 15.6.0
npm ERR! argv "/usr/local/Cellar/node/6.7.0/bin/node" "/usr/local/bin/npm" "i"
npm ERR! node v6.7.0
npm ERR! npm  v3.10.7
npm ERR! path /Users/jkamenik/github/minimap/node_modules/js-beautify/js/bin/css-beautify.js
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall chmod

npm ERR! enoent ENOENT: no such file or directory, chmod '/Users/jkamenik/github/minimap/node_modules/js-beautify/js/bin/css-beautify.js'
npm ERR! enoent ENOENT: no such file or directory, chmod '/Users/jkamenik/github/minimap/node_modules/js-beautify/js/bin/css-beautify.js'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/jkamenik/github/minimap/npm-debug.log
```

Here is the log file

```
50342 error Darwin 15.6.0
50343 error argv "/usr/local/Cellar/node/6.7.0/bin/node" "/usr/local/bin/npm" "i"
50344 error node v6.7.0
50345 error npm  v3.10.7
50346 error path /Users/jkamenik/github/minimap/node_modules/js-beautify/js/bin/css-beautify.js
50347 error code ENOENT
50348 error errno -2
50349 error syscall chmod
50350 error enoent ENOENT: no such file or directory, chmod '/Users/jkamenik/github/minimap/node_modules/js-beautify/js/bin/css-beautify.js'
50351 error enoent ENOENT: no such file or directory, chmod '/Users/jkamenik/github/minimap/node_modules/js-beautify/js/bin/css-beautify.js'
50351 error enoent This is most likely not a problem with npm itself
50351 error enoent and is related to npm not being able to find a file.
50352 verbose exit [ -2, true ]
```

I am able to reproduce the issue on command by doing the following:

1. Open an new empty file
    1. Don't save the file or issue less likely to happen
1. Paste either a very long line with soft wrap enabled, or several pages of text
1. Right click tab and Split left/right
1. Error is presented
1. Click anywhere on the document, Error is presented again
1. Double Click (Highlight) anywhere on document, Error is presented again

Minimap has to be active at the time of the split.

At this point the only way to recover is to restart Atom.

--- a/lib/adapters/stable-adapter.js
+++ b/lib/adapters/stable-adapter.js
@@ -57,6 +57,12 @@ module.exports = class StableAdapter {
     const scrollTop = this.textEditorElement.getScrollTop()
     const lineHeight = this.textEditor.getLineHeightInPixels()
     let firstRow = this.textEditorElement.getFirstVisibleScreenRow()
+    
+    if (Number.isNaN(firstRow)) {
+      // Guard against their being no visible screen row
+      return 0
+    }
+    
     let lineTop = this.textEditorElement.pixelPositionForScreenPosition([firstRow, 0]).top
 
     if (lineTop > scrollTop) {

